arch/x86/core/ia32/gdbstub.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_internal.h>
+#include <ia32/exception.h>
+#include <inttypes.h>
+#include <debug/gdbstub.h>
+
+
+extern int z_gdb_main_loop(struct gdb_ctx *ctx, bool start);
+
+static struct gdb_ctx ctx;
+static bool start;
+
+/**
+ * Currently we just handle vectors 1 and 3 but lets keep it generic
+ * to be able to notify other exceptions in the future
+ */
+static unsigned int getException(unsigned int vector)
+{
+	unsigned int exception;
+
+	switch (vector) {
+	case 0:
+		exception = 8;
```

Comment:
We have macros for all the exception IDs in arch/x86/include/kernel_arch_data.h please consider using them instead of hard-coded values here.

Reply:
also need some kind of header defining the values you are setting to. in general avoid raw numerical values as much as we can in code